tools/permission_prompt_test.py

```diff
@@ -5,220 +5,189 @@
         os.close(fd)  # can't do it sooner: it leads to errno.EIO error
     p.wait()
     return p.returncode, res['stdout'], res['stderr']
 
 
+def wrap_test(test_name, test_method, *argv):
+    sys.stdout.write(test_name + " ... ")
+    try:
+        test_method(*argv)
+        print green_ok()
+    except AssertionError:
+        print red_failed()
+        raise
+
+
 class Prompt(object):
-    def __init__(self, deno_exe):
+    def __init__(self, deno_exe, test_types):
         self.deno_exe = deno_exe
+        self.test_types = test_types
 
-    def run(self,
-            arg,
-            bytes_input,
-            allow_read=False,
-            allow_write=False,
-            allow_net=False,
-            allow_env=False,
-            allow_run=False,
-            no_prompt=False):
+    def run(self, arg, bytes_input, args=None):
         "Returns (return_code, stdout, stderr)."
-        cmd = [self.deno_exe, PERMISSIONS_PROMPT_TEST_TS, arg]
-        if allow_read:
-            cmd.append("--allow-read")
-        if allow_write:
-            cmd.append("--allow-write")
-        if allow_net:
-            cmd.append("--allow-net")
-        if allow_env:
-            cmd.append("--allow-env")
-        if allow_run:
-            cmd.append("--allow-run")
-        if no_prompt:
-            cmd.append("--no-prompt")
+        if args is None:
+            args = []
+        cmd = [self.deno_exe, PERMISSIONS_PROMPT_TEST_TS, arg] + args
         return tty_capture(cmd, bytes_input)
 
     def warm_up(self):
         # ignore the ts compiling message
-        self.run('needsWrite', b'', allow_write=True)
-
-    def test_read_yes(self):
-        code, stdout, stderr = self.run('needsRead', b'y\n')
+        self.run('needsWrite', b'', ["--allow-write"])
+
+    def test(self):
+        for test_type in self.test_types:
+            test_name_base = "test_" + test_type
+            wrap_test(test_name_base + "_allow_flag", self.test_allow_flag,
+                      test_type)
+            wrap_test(test_name_base + "_yes_yes", self.test_yes_yes,
+                      test_type)
+            wrap_test(test_name_base + "_yes_no", self.test_yes_no, test_type)
+            wrap_test(test_name_base + "_no_no", self.test_no_no, test_type)
+            wrap_test(test_name_base + "_no_yes", self.test_no_yes, test_type)
+            wrap_test(test_name_base + "_allow", self.test_allow, test_type)
+            wrap_test(test_name_base + "_deny", self.test_deny, test_type)
+            wrap_test(test_name_base + "_unrecognized_option",
+                      self.test_unrecognized_option, test_type)
+            wrap_test(test_name_base + "_no_prompt", self.test_no_prompt,
+                      test_type)
+            wrap_test(test_name_base + "_no_prompt_allow",
+                      self.test_no_prompt_allow, test_type)
+
+    def test_allow_flag(self, test_type):
+        code, stdout, stderr = self.run("needs" + test_type.capitalize(), b'',
```

Comment:
I wonder if it's worth have self.run take test_type as it's first argument.

Reply:
That would be a little cleaner, but we might need to implement tests that don't follow this pattern in the future. I do think that I should combine `arg` and `args` now that I'm looking at it again.